### PR TITLE
axum backend for new http api

### PR DESCRIPTION
**Pull request by al8n**

### CODE REVIEW

File: nomos-http-api/Cargo.toml [hunk 1]
```diff
@@ -3,20 +3,29 @@ name = "nomos-http-api"
...
+axum = ["dep:axum", "dep:hyper", "utoipa-swagger-ui/axum"]
 
 [dependencies]
 async-trait = "0.1"
+bytes = "1"
 overwatch-rs = { git = "https://github.com/logos-co/Overwatch", rev = "6e6678b" }
 overwatch-derive = { git = "https://github.com/logos-co/Overwatch", rev = "6e6678b" }
 tracing = "0.1"
+serde = { version = "1", features = ["derive"] }
+tokio = { version = "1.32", features = ["full"] }
```
Comment (zeegomo) [inline nomos-http-api/Cargo.toml:16]: nit: lazy? :)

File: nomos-http-api/src/http/backend/axum.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+use axum::{Router, Server};
+use utoipa::OpenApi;
+use utoipa_swagger_ui::SwaggerUi;
+
+use crate::Backend;
+
+#[derive(Clone)]
+pub struct AxumBackendSettings {
+    pub da: OverwatchHandle,
```
Comment (zeegomo) [inline nomos-http-api/src/http/backend/axum.rs:12]: what's this field for?

Comment (al8n) [reply nomos-http-api/src/http/backend/axum.rs:12]: Ah, this is a left-over when splitting the PR.

File: nomos-http-api/Cargo.toml [hunk 3]
```diff
@@ -27,5 +27,5 @@ axum = "0.6"
 hyper = { version = "0.14", features = ["full"] }
 utoipa-swagger-ui = { version = "4.0", features = ["axum"] }
 reqwest = { version = "0.11", features = ["blocking", "json"] }
-serde = { version = "1", features = ["derive"] }
 serde_json = "1"
+tokio = { version = "1.32", features = ["full"] }
```
Comment (zeegomo) [inline nomos-http-api/Cargo.toml:31]: why's tokio added as a dev dependency here? I don't see this PR adding any test 🤔

Comment (al8n) [reply nomos-http-api/Cargo.toml:31]: Ah, I remember why this was added, previously, it is actually a left-over, previously, I tried to use `#[tokio::test]` in test cases.

File: nomos-http-api/Cargo.toml [hunk 4]
```diff
@@ -3,20 +3,28 @@ name = "nomos-http-api"
 version = "0.1.0"
 edition = "2021"
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+[features]
+axum = ["dep:axum", "dep:hyper", "utoipa-swagger-ui/axum"]
 
 [dependencies]
 async-trait = "0.1"
+bytes = "1"
```
Comment (zeegomo) [inline nomos-http-api/Cargo.toml:11]: nit: is this used?

File: nomos-http-api/Cargo.toml [hunk 5]
```diff
@@ -3,20 +3,28 @@ name = "nomos-http-api"
...
 overwatch-rs = { git = "https://github.com/logos-co/Overwatch", rev = "6e6678b" }
 overwatch-derive = { git = "https://github.com/logos-co/Overwatch", rev = "6e6678b" }
 tracing = "0.1"
+serde = { version = "1", features = ["derive"] }
+
+thiserror = "1"
```
Comment (zeegomo) [inline nomos-http-api/Cargo.toml:17]: same

**Pull request merged**